widgetry/src/canvas.rs

```diff
@@ -353,6 +358,30 @@ impl Canvas {
         };
         ScreenPt::new(x1, y1)
     }
+
+    pub fn is_unzoomed(&self) -> bool {
+        self.cam_zoom < self.settings.min_zoom_for_detail
+    }
+
+    pub fn is_zoomed(&self) -> bool {
+        self.cam_zoom >= self.settings.min_zoom_for_detail
+    }
+
+    pub(crate) fn is_dragging(&self) -> bool {
+        // This could be called before or after handle_event. So we need to repeat the threshold
```

Comment:
This is really gross and weird. `runner` is already doing some work to figure out when a drag ends. We should figure out how to consolidate/simplify this. :(